client/components/drop-zone/index.jsx

```diff
@@ -1,81 +1,71 @@
 /**
  * External dependencies
  */
-var ReactDom = require( 'react-dom' ),
-	React = require( 'react' ),
-	createFragment = require( 'react-addons-create-fragment' ),
-	without = require( 'lodash/without' ),
-	includes = require( 'lodash/includes' ),
-	classNames = require( 'classnames' ),
-	noop = require( 'lodash/noop' );
+import ReactDom from 'react-dom';
+import React, { Component, PropTypes } from 'react';
+import createFragment from 'react-addons-create-fragment';
+import without from 'lodash/without';
+import includes from 'lodash/includes';
+import classNames from 'classnames';
+import noop from 'lodash/noop';
+import identity from 'lodash/identity';
 
 /**
  * Internal dependencies
  */
-var RootChild = require( 'components/root-child' );
+import RootChild from 'components/root-child';
+import localize from 'lib/mixins/i18n/localize';
 
-module.exports = React.createClass( {
-	displayName: 'DropZone',
+export class DropZone extends Component {
+	constructor( props ) {
+		super( props );
 
-	propTypes: {
-		onDrop: React.PropTypes.func,
-		onVerifyValidTransfer: React.PropTypes.func,
-		onFilesDrop: React.PropTypes.func,
-		fullScreen: React.PropTypes.bool,
-		icon: React.PropTypes.string
-	},
-
-	getInitialState: function() {
-		return {
+		this.state = {
 			isDraggingOverDocument: false,
 			isDraggingOverElement: false
 		};
-	},
-
-	getDefaultProps: function() {
-		return {
-			onDrop: noop,
-			onVerifyValidTransfer: () => true,
-			onFilesDrop: noop,
-			fullScreen: false,
-			icon: 'dashicons dashicons-admin-media'
-		};
-	},
 
-	componentDidMount: function() {
+		// Bind listeners found in componentDidMount(), except for the ones whose
+		// implementation doesn't refer to `this`.
+		this.onDrop = this.onDrop.bind( this );
```

Comment:
>any practical examples for using the unbound methods you can think of?

No, not off the top of my head. And even if there were, I suppose it could be as simple as applying another `bind` to the already-bound function?

I wasn't aware of the React documentation including that as an example, and you make a good point that this replicates the behavior of `React.createClass`. Then again, one could argue that the behavior of `React.createClass` is not one that should be replicated 😄  (given that it's not obvious at a glance how the functions are bound)